### PR TITLE
Add accessors to Consume

### LINKED ISSUE #145: External code can't access Consume fields, and thus can't implement Consumer
I ran into this when copying over the implementation from #137 into a crate outside of `rubble` - it looks like `Consume`, the type used for conveying results to `Consumer`, has private fields only accessible to rubble. Unless I'm missing a way to get into `Consume::consume` and `Consume::result`, this prevents outside users from implementing queues built to their needs.

I think this could be fixed pretty easily by either making fields public (maybe with an adding private `()` field for forward compatibility), or by adding accessors. If it's alright, I'll submit a PR - I mildly favor accessors, but since rubble's still pre-`1.0`, and `Consume` is fairly feature-complete for what it is, it probably doesn't matter a ton?

### PR DESCRIPTION
**Pull request by daboross**
Fixes #145

I haven't thought a ton about the naming of these methods, but they should work alright.

### CODE REVIEW

File: rubble/src/link/queue.rs [hunk 1]
```diff
@@ -190,6 +190,21 @@ impl<T> Consume<T> {
...
+    pub fn consume(&self) -> bool {
+        self.consume
+    }
+
+    /// Retrieves a reference to the inner result.
+    pub fn result(&self) -> &Result<T, Error> {
+        &self.result
+    }
+
+    /// Unwraps this [`Consume`], retrieving the inner result.
```
Comment (jonas-schievink) [inline rubble/src/link/queue.rs:204]: The link doesn't go anywhere

File: rubble/src/link/queue.rs [hunk 2]
```diff
@@ -190,6 +190,21 @@ impl<T> Consume<T> {
             result,
         }
     }
+
+    /// Retrieves whether this consume represents consuming a packet or not.
```
Comment (jonas-schievink) [inline rubble/src/link/queue.rs:194]:
```suggestion
    /// Retrieves whether the packet should be removed from the queue.
```

File: rubble/src/link/queue.rs [hunk 3]
```diff
@@ -190,6 +190,21 @@ impl<T> Consume<T> {
...
+    pub fn consume(&self) -> bool {
```
Comment (jonas-schievink) [inline rubble/src/link/queue.rs:195]: Maybe `should_consume` would be better? Feel free to rename the field to match as well.

Comment (daboross) [reply rubble/src/link/queue.rs]: Renamed!

File: rubble/src/link/queue.rs (hunk 1 above)
Comment (daboross) [reply rubble/src/link/queue.rs]: Right. I'm not entirely sure what I was thinking with this, thanks for catching it!

**Pull request merged**